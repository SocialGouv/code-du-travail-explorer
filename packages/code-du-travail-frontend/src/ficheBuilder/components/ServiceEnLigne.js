import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { childrenBuilder } from "../index";
import cssVars from "../cssVariables";

const ServiceEnLigne = ({ data }) => {
  const type = data._.type;
  const url = data._.URL;
  const title = childrenBuilder(data.$[0]);
  const source = childrenBuilder(data.$[1]);
  return (
    <Wrapper>
      <Type>{type}</Type>
      <ButtonLink href={url} target="_blank">
        {`${title}`}
      </ButtonLink>
      <Source>{source}</Source>
    </Wrapper>
  );
};

ServiceEnLigne.propTypes = {
  data: PropTypes.object.isRequired
};

export default ServiceEnLigne;

const Wrapper = styled.div`
  display: flex;
  flex-flow: column;
  align-items: center;
  padding: ${cssVars.spacing};
  background-color: ${cssVars.veryLightGrey};
  border-radius: ${cssVars.borderRadius};
`;

const Type = styled.div`
  align-self: flex-start;
  margin-bottom: ${cssVars.lightSpacing};
`;

const ButtonLink = styled.a`
  &:link,
  &:visited {
    display: inline-block;
    background-color: ${cssVars.blue};
    color: ${cssVars.white};
    text-decoration: none;
    padding: ${cssVars.buttonThickness};
    border-radius: ${cssVars.borderRadius};
  }
  &:hover,
  &:focus,
  &:active {
    background-color: ${cssVars.darkBlue};
  }
`;

const Source = styled.small`
  display: inline-block;
  margin-top: ${cssVars.minimalSpacing};
  color: ${cssVars.lightGrey};
`;
