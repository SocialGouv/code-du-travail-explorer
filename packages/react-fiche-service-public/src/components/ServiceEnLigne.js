import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { theme } from "@cdt/ui";

import { getText } from "../utils";

const { colors, spacing, box } = theme;

class ServiceEnLigne extends React.PureComponent {
  static propTypes = {
    data: PropTypes.object.isRequired
  };
  render() {
    const { data } = this.props;
    const type = data._.type;
    const url = data._.URL;
    const title = getText(data.$[0]);
    const source = getText(data.$[1]);
    return (
      <Wrapper>
        <Type>{type}</Type>
        <ButtonLink href={url} rel="noopener noreferrer" target="_blank">
          {title}
        </ButtonLink>
        <Source>{source}</Source>
      </Wrapper>
    );
  }
}

export default ServiceEnLigne;

const Wrapper = styled.div`
  display: flex;
  flex-flow: column;
  align-items: center;
  margin-bottom: ${spacing.base};
  padding: ${spacing.large};
  background-color: ${colors.darkBackground};
  border-radius: ${box.borderRadius};
`;

const Type = styled.div`
  align-self: flex-start;
  margin-bottom: ${spacing.xsmall};
  font-weight: bold;
  font-size: 1.1rem;
  color: ${colors.title};
`;

const ButtonLink = styled.a`
  &:link,
  &:visited {
    display: inline-block;
    background-color: ${colors.primaryBackground};
    color: ${colors.primaryText};
    text-decoration: none;
    padding: ${spacing.xsmall};
    border-radius: ${box.borderRadius};
  }
  &:hover,
  &:focus,
  &:active {
    background-color: ${colors.blueDark};
  }
`;

const Source = styled.small`
  display: inline-block;
  margin-top: ${spacing.xsmall};
  color: ${colors.darkGrey};
`;
