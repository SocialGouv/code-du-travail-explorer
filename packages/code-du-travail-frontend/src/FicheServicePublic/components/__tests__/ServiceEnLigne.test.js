import React from "react";
import { render } from "react-testing-library";
import ServiceEnLigne from "../ServiceEnLigne";
import serviceEnLigneDataMock from "./mocks/serviceEnLigneData.json";

describe("<ServiceEnLigne />", () => {
  it("should render", () => {
    const { container } = render(
      <ServiceEnLigne data={serviceEnLigneDataMock} />
    );
    expect(container).toMatchSnapshot();
  });
});
