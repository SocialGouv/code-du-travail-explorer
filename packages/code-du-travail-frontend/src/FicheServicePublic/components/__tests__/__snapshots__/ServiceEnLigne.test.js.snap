// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServiceEnLigne /> should render 1`] = `
<div>
  <div
    class="sc-htpNat ioqBgE"
  >
    <div
      class="sc-bxivhb kjpcZS"
    >
      Se retrouve dans en entête
    </div>
    <a
      class="sc-ifAKCX iPrfNR"
      href="url à retoruver dans le lien"
      rel="noopener noreferrer"
      target="_blank"
    >
      Texte qui se retrouve dans le lien
    </a>
    <small
      class="sc-EHOje jrDJBX"
    >
      Source qui se retrouve sous le lien
    </small>
  </div>
</div>
`;
